sycl/include/CL/sycl/image.hpp

```diff
@@ -105,40 +193,47 @@ class image {
 
   bool operator!=(const image &rhs) const { return !(*this == rhs); }
 
-  /* -- common interface members -- */
-
   /* -- property interface members -- */
+  template <typename propertyT> bool has_property() const {
+    return impl->template has_property<propertyT>();
+  }
 
-  range<dimensions> get_range() const { return impl->get_range(); }
+  template <typename propertyT> propertyT get_property() const {
+    return impl->template get_property<propertyT>();
+  }
 
-  /* Available only when: dimensions > 1 */
-  range<dimensions - 1> get_pitch() const { return impl->get_pitch(); }
+  range<Dimensions> get_range() const { return impl->get_range(); }
 
+  /* Available only when: dimensions >1 */
+  template <bool B = (Dimensions > 1)>
+  typename std::enable_if<B, range<Dimensions - 1>>::type get_pitch() const {
+    return impl->get_pitch();
+  }
+
+  // Returns the size of the image storage in bytes
   size_t get_size() const { return impl->get_size(); }
 
+  // Returns the total number of elements in the image
   size_t get_count() const { return impl->get_count(); }
 
+  // Returns the allocator provided to the image
   AllocatorT get_allocator() const { return impl->get_allocator(); }
 
-  template <typename dataT, access::mode accessMode>
-  accessor<dataT, dimensions, accessMode, access::target::image>
-  get_access(handler &commandGroupHandler) {
-    return impl->template get_access<dataT, accessMode>();
+  template <typename Destination = std::nullptr_t>
+  void set_final_data(Destination FinalData = nullptr) {
+    if (true)
+      throw cl::sycl::feature_not_supported("Feature Not Implemented");
+    return;
   }
 
-  template <typename dataT, access::mode accessMode>
-  accessor<dataT, dimensions, accessMode, access::target::host_image>
-  get_access() {
-    return impl->template get_access<dataT, accessMode>();
+  void set_write_back(bool Flag = true) {
+    if (true)
+      throw cl::sycl::feature_not_supported("Feature Not Implemented");
+    return;
   }
 
-  //template <typename Destination = std::nullptr_t>
-  //void set_final_data(Destination finalData = std::nullptr);
-
-  void set_write_back(bool flag = true) { impl->set_write_back(flag); }
-
 private:
-  shared_ptr_class<detail::image_impl<dimensions, AllocatorT>> impl;
+  shared_ptr_class<detail::image_impl<Dimensions, AllocatorT>> impl;
   template <class Obj>
   friend decltype(Obj::impl) detail::getSyclObjImpl(const Obj &SyclObject);
```

Comment:
You probably need to apply similar changes like in this patch https://github.com/intel/llvm/pull/217